channel.go

```diff
@@ -340,7 +347,10 @@ func (ch *Channel) dispatch(msg message) {
 		// deliveries are in flight and a no-wait cancel has happened
 
 	default:
-		ch.rpc <- msg
+		select {
+		case ch.rpc <- msg:
+		default:
```

Comment:
Same as on the connection, this seems like it could drop replies for unintended reasons.